microservices/recordings/src/Recording/Encoder.php

```diff
@@ -109,117 +119,149 @@ public function processAction()
 
             // Get callid from file
             $this->logger->info(sprintf("[Recordings][%s] Checking file %s\n", $hashid, $file));
 
-            // Look if the converstation with that id has ended
-            /** @var TrunksCdrInterface $kamAccCdr */
-            $kamAccCdr = $this->trunksCdrRepository->findOneByCallid($callid);
-            if ($kamAccCdr) {
-                $type = 'ddi';
-                if ($kamAccCdr->getXcallid()) {
-                    // If call first leg, caller is who activated the recording
-                    $recorder = $kamAccCdr->getCaller();
-                } else {
-                    // If call second leg, callee is who activated the recording
-                    $recorder = $kamAccCdr->getCallee();
-                }
-            } else {
-                $kamAccCdr = $this->usersCdrRepository->findOneByCallid($callid);
-                if ($kamAccCdr) {
-                    $type = 'ondemand';
-                    if ($kamAccCdr->getXcallid()) {
-                        // If call second leg, callee is who activated the recording
-                        $recorder = $kamAccCdr->getCallee();
-                    } else {
+            // Look if the conversation with that id has ended
+            $kamAccCdrs = [];
+            $kamAccCdrs = array_merge($kamAccCdrs, $this->trunksCdrRepository->findBy(['callid' => $callid]));
+            if (empty($kamAccCdrs)) {
+                $kamAccCdrs = array_merge($kamAccCdrs, $this->usersCdrRepository->findBy(['callid' => $callid]));
+            }
+
+            if (empty($kamAccCdrs)) {
+                $stats['skipped']++;
+                $this->logger->info(sprintf("[Recordings][%s] Call with id = %s has not yet finished!\n",
+                    $hashid, $callid));
+                continue;
+            }
+
+            foreach ($kamAccCdrs as $kamAccCdr) {
+
+                if ($kamAccCdr instanceof TrunksCdrInterface) {
+                    $type = 'ddi';
+                    $direction = $kamAccCdr->getDirection();
+
+                    if ($direction == 'outbound') {
                         // If call first leg, caller is who activated the recording
                         $recorder = $kamAccCdr->getCaller();
+                    } else {
+                        // If call second leg, callee is who activated the recording
+                        $recorder = $kamAccCdr->getCallee();
+                    }
+
+                    // Check this ddi has recording enabled
+                    $ddi = $this->ddiRepository->findOneByDdiE164($recorder);
+                    if (!$ddi) {
+                        $this->logger->error(
+                            sprintf("[Recordings][%s] Unable to find DDI for %s\n", $hashid, $recorder)
+                        );
+                        continue;
+                    }
+
+                    if (!in_array($ddi->getRecordCalls(), array('all', $direction))) {
```

Comment:
php's in_array is quite misleading ¿Could you set strict flag here? more info http://php.net/manual/es/function.in-array.php#61491